### PR TITLE
Changed gendered terms to be gender-neutral

### PR DESCRIPTION
**Pull request by gonyere**
Just going through and updating terms to be gender neutral. Also changed an instance of 'he' to 'be' where it seemed appropriate, and one of 'his' to 'this' - both where it seemed they were typos.

### CODE REVIEW

File: docs/CODING_STYLE.md [hunk 1]
```diff
@@ -297,7 +297,7 @@ SPDX-License-Identifier: LGPL-2.1-or-later
   with a more brutal `assert()`. We are more forgiving to public users than for
   ourselves! Note that `assert()` and `assert_return()` really only should be
   used for detecting programming errors, not for runtime errors. `assert()` and
-  `assert_return()` by usage of `_likely_()` inform the compiler that he should
+  `assert_return()` by usage of `_likely_()` inform the compiler that they should
```
Comment (bluca) [inline docs/CODING_STYLE.md]: it's referring to the compiler, so `it` would be right here I think?

**Review by bluca — APPROVED** [review]
just one nit, otherwise looks good, thanks!

File: src/libsystemd/sd-bus/bus-socket.c [hunk 2]
```diff
@@ -302,7 +302,7 @@ static int verify_external_token(sd_bus *b, const char *p, size_t l) {
         int r;
 
         /* We don't do any real authentication here. Instead, we if
```
Comment (keszybz) [inline src/libsystemd/sd-bus/bus-socket.c]: "we if" — please also fix this.

File: man/sd_bus_add_object.xml [hunk 3]
```diff
@@ -508,7 +508,7 @@
         <varlistentry>
           <term><constant>SD_BUS_VTABLE_METHOD_NO_REPLY</constant></term>
 
-          <listitem><para>Mark his vtable entry as a method that will not return a reply using the
```
Comment (mbiebl) [inline man/sd_bus_add_object.xml:511]: this is technically a typo fix

Comment (bluca) [reply man/sd_bus_add_object.xml:511]: it's ok, we often update/reformat many docs in the same commit when doing fixups like these

**Pull request merged**